lib/puppet-lint/plugins/check_unsafe_interpolations/check_unsafe_interpolations.rb

```diff
@@ -0,0 +1,130 @@
+      # Initialise variables needed in while loop
+      rule_violations = []
+      current_token = token
+
+      # Iterate through tokens in command
+      while current_token.type != :NEWLINE
+      # Check if token is a varibale and if it is parameterised
+      rule_violations.append(current_token.next_code_token) if interpolated?(current_token)
+      current_token = current_token.next_token
+      end
+
+      rule_violations
+  end
+
+  # A command is parameterised if its args are placed in an array
+  # This function checks if the current token is a :FARROW and if so, if it is followed by an LBRACK
+  def parameterised?(token)
+      current_token = token
+      while current_token.type != :NEWLINE
+      return true if current_token.type == :FARROW && current_token.next_token.next_token.type == :LBRACK
+      current_token = current_token.next_token
+      end
+  end
+
+  # This function is a replacement for puppet_lint's title_tokens function which assumes titles have single quotes
+  # This function adds a check for titles in double quotes where there could be interpolated variables
+  def get_exec_titles
+      result = []
+      tokens.each_index do |token_idx|
+      next unless ['exec'].include?(tokens[token_idx].value)
+      # We have a resource declaration. Now find the title
+      tokens_array = []
+      # Check if title is an array
+      if tokens[token_idx].next_code_token.next_code_token.type == :LBRACK
+          # Get the start and end indices of the array of titles
+          array_start_idx = tokens.rindex { |r| r.type == :LBRACK }
+          array_end_idx = tokens.rindex { |r| r.type == :RBRACK }
+
+          # Grab everything within the array
+          title_array_tokens = tokens[(array_start_idx + 1)..(array_end_idx - 1)]
+          tokens_array.concat(title_array_tokens.reject do |token|
+          USELESS_CHARS.include?(token.type)
+          end)
+          result << tokens_array
+      # Check if title is double quotes string
+      elsif tokens[token_idx].next_code_token.next_code_token.type == :DQPRE
+          # Find the start and end of the title
+          title_start_idx = tokens.find_index(tokens[token_idx].next_code_token.next_code_token)
+          title_end_idx = title_start_idx + index_offset_for(':', tokens[title_start_idx..tokens.length])
+
+          result << tokens[title_start_idx..title_end_idx]
+      # Title is in single quotes
+      else
+          tokens_array.concat([tokens[token_idx].next_code_token.next_code_token])
+
+          result << tokens_array
+      end
+      end
+      result
+  end
+
+  def interpolated?(token)
+      INTERPOLATED_STRINGS.include?(token.type)
+  end
+
+  # Finds the index offset of the next instance of `value` in `tokens_slice` from the original index
+  def index_offset_for(value, tokens_slice)
+      i = 0
+      while i < tokens_slice.length
+      return i if value.include?(tokens_slice[i].value)
+      i += 1
+      end
+  end
+end
+  
```

Comment:
```suggestion
```

Reply:
wow, Rubocop really isn't happy with me... thanks @ekohl. I'll fix this up

Reply:
also, what is PUP-5704? Do you mean a specific version of puppet?